From e3232bb3d0649f3fdce01091e1948fb8e906f9eb Mon Sep 17 00:00:00 2001
From: Matthew Leeds <matthew.leeds@endlessm.com>
Date: Mon, 3 Feb 2020 11:24:41 -0800
Subject: [PATCH] portal: Add g_autoptr() defines for old GLib versions

We don't need to check for GLib 2.44 (the first release with g_autoptr()
support) since Flatpak requires that version in configure.ac.

Fixes https://github.com/flatpak/flatpak/issues/3403
---
 portal/flatpak-portal.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/portal/flatpak-portal.c b/portal/flatpak-portal.c
index aa0d2d399..060987b7f 100644
--- a/portal/flatpak-portal.c
+++ b/portal/flatpak-portal.c
@@ -50,6 +50,14 @@
 #include "portal-impl.h"
 #include "flatpak-permission-dbus.h"
 
+/* GLib 2.47.92 was the first release to define these in gdbus-codegen */
+#if !GLIB_CHECK_VERSION (2, 47, 92)
+G_DEFINE_AUTOPTR_CLEANUP_FUNC (PortalFlatpakProxy, g_object_unref)
+G_DEFINE_AUTOPTR_CLEANUP_FUNC (PortalFlatpakSkeleton, g_object_unref)
+G_DEFINE_AUTOPTR_CLEANUP_FUNC (PortalFlatpakUpdateMonitorProxy, g_object_unref)
+G_DEFINE_AUTOPTR_CLEANUP_FUNC (PortalFlatpakUpdateMonitorSkeleton, g_object_unref)
+#endif
+
 #define IDLE_TIMEOUT_SECS 10 * 60
 
 static GHashTable *client_pid_data_hash = NULL;
